File: 3351-Maximize Happiness of Selected Children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        n = len(happiness)
        happiness.sort()
        res = 0
        for i in range(n-1, n-k-1, -1):
            rnds = n - i - 1
            #print(rnds, i, n)
            res += max(happiness[i] - rnds,0)
        
        return res
        

File: 3351-Maximize Happiness of Selected Children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        //System.out.println(Arrays.toString(happiness));
        long res = 0;
        int n = happiness.length;
        for(int i = n -1; i > n - k - 1; i--){
            int rnds = n - i - 1;
            res += Math.max(happiness[i] - rnds, 0); 
        }
        return res;
    }
}